### PR TITLE
Clean up various things discovered by D-Scanner

### PR DESCRIPTION
**Pull request by Hackerpilot**

### CODE REVIEW

File: std/experimental/allocator/quantizer.d [hunk 1]
```diff
@@ -213,21 +213,21 @@ struct Quantizer(ParentAllocator, alias roundingFunction)
 ///
 unittest
 {
-    import std.experimental.allocator.free_tree,
-        std.experimental.allocator.gc_allocator;
+    import std.experimental.allocator.free_tree : FreeTree;
+	import std.experimental.allocator.gc_allocator : GCAllocator;
```
Comment (CyberShadow) [inline std/experimental/allocator/quantizer.d]: Whitespace looks weird here

File: std/experimental/allocator/package.d [hunk 2]
```diff
@@ -231,13 +231,13 @@ in turn uses the garbage collected heap.
 unittest
 {
     // Install a new allocator that is faster for 128-byte allocations.
-    import std.experimental.allocator.free_list,
-        std.experimental.allocator.gc_allocator;
+    import std.experimental.allocator.free_list : FreeList;
+	import std.experimental.allocator.gc_allocator : GCAllocator;
```
Comment (CyberShadow) [inline std/experimental/allocator/package.d]: And here

**Comment by Hackerpilot** [comment]
Reminder: FreeList.min doesn't use its "low" parameter.

**Comment by andralex** [comment]
I think `factory` is best left public. I'll pull your work then perhaps adjust a couple things. Thanks very much!

File: std/experimental/allocator/common.d [hunk 3]
```diff
@@ -47,9 +47,9 @@ struct Ternary
       $(TR $(TD `no`) $(TD `unknown`) $(TD) $(TD `unknown`) $(TD `no`) $(TD `unknown`))
       $(TR $(TD `yes`) $(TD `no`) $(TD `no`) $(TD `yes`) $(TD `no`) $(TD `yes`))
       $(TR $(TD `yes`) $(TD `yes`) $(TD) $(TD `yes`) $(TD `yes`) $(TD `no`))
-      $(TR $(TD `yes`) $(TD `unknown`) $(TD) $(TD `yes`) $(TD `unknown`) $(TD `unknown`))
+      $(TR $(TD `yes`) $(TD `unknown`) $(TD) $(TD `yes`) $(TD `no`) $(TD `unknown`))
```
Comment (andralex) [inline std/experimental/allocator/common.d]: Hmm, here it seems you undid my other commit...

File: std/experimental/allocator/free_list.d [hunk 4]
```diff
@@ -542,7 +542,7 @@ struct ContiguousFreeList(ParentAllocator,
 
     /// ditto
     static if (!stateSize!ParentAllocator
-        && maxSize == chooseAtRuntime || maxSize == unbounded
+        && (maxSize == chooseAtRuntime || maxSize == unbounded)
```
Comment (andralex) [inline std/experimental/allocator/free_list.d]: wow, that was a bug!

Comment (Hackerpilot) [inline std/experimental/allocator/free_list.d]: I thought this didn't actually change any thing since || has higher precedence than &&.

Comment (CyberShadow) [inline std/experimental/allocator/free_list.d]: Umm, I have news for you.

http://wiki.dlang.org/Operator_precedence

(Useful mnemonic: `&&` is "heaver" and "pulls" sub-expressions harder than `||`)

Comment (andralex) [inline std/experimental/allocator/free_list.d]: OK, so it was a double bug: in my code and in dfix!!!

Comment (andralex) [inline std/experimental/allocator/free_list.d]: (my mnemonic: || is like addition and && is like multiplication)

Comment (Hackerpilot) [inline std/experimental/allocator/free_list.d]: That wasn't dfix. I did that manually.

Comment (Hackerpilot) [inline std/experimental/allocator/free_list.d]: I found out what happened. When I looked at the grammar I confused "OrExpression" and "OrOrExpression".

Comment (CyberShadow) [inline std/experimental/allocator/free_list.d]: `a && b || c` should totally be a dfix rule (or even a compiler warning) though.

**Pull request merged**